rhoc/pkg/cmd/connectors/patch/type.go

```diff
@@ -0,0 +1,41 @@
+package patch
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/cos-tools/rhoc/pkg/service"
+	"github.com/bf2fc6cc711aee1a0c2a/cos-tools/rhoc/pkg/util/response"
+	"github.com/redhat-developer/app-services-cli/pkg/core/ioutil/dump"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+)
+
+type options struct {
+	id           string
+	skipConfirm  bool
+	outputFormat string
+
+	f *factory.Factory
+}
+
+func execute(opts options, content map[string]interface{}) error {
```

Comment:
Can we have options own this method? It would need to change the way the method is called in other usnits

```suggestion
func (opts options) execute(content map[string]interface{}) error {
```

Reply:
we can but, what is the rationale ?

Reply:
oh it really doesn't make a difference for such a small package, I am fine either way it is a just a consideration

Reply:
In a package with more complexity you want to follow a model where the concrete data exhibits a behaviour instead of having a bunch of functions. At that point you probably want to define interfaces that describe those behaviours. 

For this very case you could ask yourself, what is this package doing. Well, it patches a connector. So we may have a data type `options`(arguably not the best name) and an interface `patcher`. `options` would behave like a `patcher` by implementing the `patch` method.

```
type options struct {...}

type patcher interface {
  patch(content map[string]interface{}) error
}

func (opts options) patch(content map[string]interface{}) error {...}
```

As a package gains in complexity, this models add clarity. Dealing with interfaces is better than dealing with the data type, you want to expose a package functionality and data behaviour instead of data types.

But for such a small package it doesn't matter. `options` is used internally anyways no much point on defining a interface neither. Personally, I rather follow a package model that fits complex scenarios as well simple scenarios(like this one).

Reply:
> 
> As a package gains in complexity, this models add clarity. Dealing with interfaces is better than dealing with the data type, you want to expose a package functionality and data behaviour instead of data types.
> 

In this case the `options` thing is really some sugar to group together stuffs and does not really model any useful data model abstraction i.e. the output format is a trait of the command, not the patch.

We can even argue if is really needed :)

Reply:
my thoughts too!!